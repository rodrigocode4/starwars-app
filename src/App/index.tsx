import React, { useEffect, useState } from 'react'
import Layout from 'Layout'
import { FunctionContext } from 'Utils/context'
import type { Character } from 'Utils/types'

function App() {
  const [search, setSearch] = useState<string>('')
  const [state, setState] = useState<Character[]>()

  useEffect(() => {
    async function getData() {
      if (search !== '') {
        const rs = await fetch(`https://swapi.dev/api/people/?search=${search}`)
        const { results } = await rs.json()

        const data: Array<Character> = []
        for await (const { name, homeworld, gender, url } of results) {
          const rs = await fetch(homeworld)
          const { name: planet } = await rs.json()
          const dataFiltered: Character = {
            name,
            homeworld: planet,
            gender,
            url
          }
          data.push(dataFiltered)
        }
        setState(data)
      }
    }
    getData()
  }, [search])

  function handleClick(searchText: string) {
    setSearch('')
    setSearch(searchText.trim())
  }

  return (
    <>
      <FunctionContext.Provider value={handleClick}>
        <Layout characters={state} />
      </FunctionContext.Provider>
    </>
  )
}

export default App
