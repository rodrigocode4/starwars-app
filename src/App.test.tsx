import React from 'react'
import { render, screen } from '@testing-library/react'

import App from './App'

describe('<App />', () => {
  test('deve renderizar App', () => {
    const { container } = render(<App />)

    const heading = screen.getByRole('heading', {name: /hello world/i})

    expect(heading).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

})

