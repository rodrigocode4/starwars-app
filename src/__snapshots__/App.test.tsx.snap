// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> deve renderizar App 1`] = `
<main>
  <h1>
    Hello World
  </h1>
</main>
`;
